crates/bevy_render/src/render_phase/mod.rs

```diff
@@ -76,7 +118,131 @@ impl<I: BatchedPhaseItem> RenderPhase<I> {
     }
 }
 
-/// This system sorts all [`RenderPhases`](RenderPhase) for the [`PhaseItem`] type.
+/// An item (entity) which will be drawn to a texture or the screen, as part of a [`RenderPhase`].
+///
+/// A phase item has to be queued up for rendering during the
+/// [`RenderStage::Queue`](crate::RenderStage::Queue).
+/// Afterwards it will be sorted and rendered automatically in the
+/// [`RenderStage::PhaseSort`](crate::RenderStage::PhaseSort) and
+/// [`RenderStage::Render`](crate::RenderStage::Render), respectively.
+pub trait PhaseItem: Sized + Send + Sync + 'static {
+    /// The type used for ordering the items. The smallest values are drawn first.
+    /// This order can be calculated using the [`ViewRangefinder3d`],
+    /// based on the view-space `Z` value of the corresponding view matrix.
+    type SortKey: Ord;
+
+    /// The corresponding entity that will be drawn.
+    fn entity(&self) -> Entity;
+
+    /// Determines the order in which the items are drawn.
+    fn sort_key(&self) -> Self::SortKey;
+
+    /// Specifies the [`Draw`] function used to render the item.
+    fn draw_function(&self) -> DrawFunctionId;
+
+    /// Sorts a slice of phase items into render order. Generally if the same type
+    /// implements [`BatchedPhaseItem`], this should use a stable sort like [`slice::sort_by_key`].
+    /// In almost all other cases, this should not be altered from the default,
+    /// which uses a unstable sort, as this provides the best balance of CPU and GPU
+    /// performance.
+    ///
+    /// Implementers can optionally not sort the list at all. This is generally advisable if and
+    /// only if the renderer supports a depth prepass, which is by default not supported by
+    /// the rest of Bevy's first party rendering crates. Even then, this may have a negative
+    /// impact on GPU-side performance due to overdraw.
```

Comment:
It's very likely that the prepass will be merged before 0.10 so we need to remember to change this when it happens depending on in what order this gets merged.

Reply:
With that said, I think even with the prepass we should keep the sorting, I don't have numbers to prove that though.

Reply:
Yeah that sounds right.
Iyes Ida was working on rearchitecting the render stages, which will require further changes to the sorting documentation aswell.